pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -584,6 +633,15 @@ func (r *ReconcileBareMetalHost) actionDeprovisioning(prov provisioner.Provision
 	// After the provisioner is done, clear the image settings so we
 	// transition to the next state.
 	info.host.Status.Provisioning.Image = metal3v1alpha1.Image{}
+	info.host.Status.Provisioning.PoweredOnAt = nil
+	info.host.Status.Provisioning.PendingRebootSince = nil
+	if info.host.Annotations != nil {
```

Comment:
This block could be a little utility function `clearRebootAnnotation`.

The results of changing the annotations are not being saved after this. Just like around like 679, we need to update the spec portion of the host here to save the annotation change.

Alternatively, if we stored enough state to know that the reboot was done, we could just leave the annotation in place and let the controller that added it remove it again. That would simplify the logic here in this controller, since we wouldn't need to use 2 different ways to store the resource.

@zaneb what do you think about that?

Reply:
> This block could be a little utility function clearRebootAnnotation.
done

> It is a problem that the annotation changes here won't get written out to the datastore when we return to Reconcile() though.
fixed